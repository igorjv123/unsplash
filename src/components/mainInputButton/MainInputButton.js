import React from 'react';
import './style.sass';
import { string, func, object } from 'prop-types';

const MainInputButton = (props) => {
    return(
        <button className={props.className} onClick={props.onClick}>{props.children}</button>
    )
}

MainInputButton.propTypes = {
    className: string,
    onClick: func,
    children: string || func || object,
};

MainInputButton.defaultProps = {
    className: 'header-button',
    onClick: () => {},
    children: 'OK'
};

export default MainInputButton;
