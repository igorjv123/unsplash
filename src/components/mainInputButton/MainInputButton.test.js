import React from 'react';
import MainInputButton from './MainInputButton';
import { shallow } from 'enzyme';

describe('MainSearch', () => {
    const onClick = jest.fn(() => 'Set new query');
    let className = 'className';
    let children = 'children';

    beforeEach(() => {
        className = 'className';
        children = 'children';
    })

    it('should call setNewQuery prop function', () => {
        const wrapper = shallow(
            <MainInputButton
                onClick = {onClick}
            >
            </MainInputButton>
        );
        const input = wrapper.find('button')

        input.simulate('click');

        expect(onClick).toHaveBeenCalled();
    });

    it('should set attribs from props', () => {
        const wrapper = shallow(
            <MainInputButton
                className = {className}
                onClick = {onClick}>
                {children}
            </MainInputButton>
        );

        expect(wrapper).toMatchSnapshot();
    });
});
