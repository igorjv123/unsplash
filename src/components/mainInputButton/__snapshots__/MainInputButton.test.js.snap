// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSearch should set attribs from props 1`] = `ShallowWrapper {}`;
